app/src/main/java/com/example/filemanager/AllFilesFragment.kt

```diff
@@ -65,8 +82,25 @@ class AllFilesFragment : Fragment(), ItemClickListener {
         fetchFiles()
     }
 
-    override fun onItemClickListener() {
-        Toast.makeText(context, "Item clicked", Toast.LENGTH_SHORT).show()
+    private fun setToolbarNavigation() {
+        binding.toolbar.setNavigationIcon(R.drawable.back_icon)
```

Comment:
Is this the recommended way to show the back button from toolbar? I haven't use the Toolbar for like more than a year and forgot how is the best way to use it.

Reply:
from multiple sources I found that we have to use the ```setNavigationIcon()``` function

Reply:
In the official documentation is given this function

Reply:
I think you can find examples of toolbar usages here 
https://guides.codepath.com/android/using-the-app-toolbar
https://developer.android.com/training/appbar/up-action
You dont need to set the navigation icon, the Toolbar should already have it.

Reply:
but, I think these examples are used for activity toolbars, but I use the toolbar in a fragment

Reply:
Ahh, so you should put it in the activity.

Reply:
but i need to handle back press in the fragment

Reply:
You can do a small research about the Toolbar API and how it works on android. You can access it from a fragment even if it is declared in the activity.